docs/syntax/pattern.md

```diff
@@ -273,26 +273,29 @@ Note that, due to the lack of an operation for distinguishing `nothing` from "ke
 Deconstruction of Custom Composite Data
 -------------------------------------------
 
-In order to deconstruct arbitrary data types in a similar way to `Tuple`, `Array` and `Dict`, simply declare them to be record types with the `@as_record` macro.
-
-Here is an example, check more about ADTs(and GADTs) at [Algebraic Data Type Syntax in MLStyle](https://thautwarm.github.io/MLStyle.jl/latest/syntax/adt).
+Enable pattern matching on the internals of a struct `S` with `@as_record S`.
 
 ```julia-console
-julia> @data Color begin
```

Comment:
It is not necessary to introduce ADTs here. `struct` is more familiar and does the job.